### PR TITLE
docs: specify 1.12.0+ent incompat with nomad

### PR DESCRIPTION
**Pull request by schmichael**
Due to build changes in Consul 1.12.0 the `+ent` modifier is missing
from the version reported by `/v1/agent/self`.

Nomad looks for the `ent` modifier when determining whether to reconcile
services in non-default namespaces. Without the modifier Nomad will only
end up removing services from the default Consul namespace.

### CODE REVIEW

**Comment by karl-cardenas-coding** [comment]
I'm tagging the tech writers to get their input. I have some concerns with the approach of how we are communicating the message.

File: website/content/docs/upgrading/upgrade-specific.mdx
Comment (trujillo-adam) [reply website/content/docs/upgrading/upgrade-specific.mdx:31]: @boruszak - I see the rationale for using a callout box, but I think we might be better served changing the heading to something like "Breaking Change for Nomad Enterprise Users" and using plain formatting. Better guidance on text formatting and callouts are some of the style guide areas that we are trying to nail down.

File: website/content/docs/upgrading/upgrade-specific.mdx
Comment (trujillo-adam) [reply website/content/docs/upgrading/upgrade-specific.mdx:34]: I think we can soften this a little more and talk about a "change" (instead of a "bug") that will result in this behavior. We can say that we recognize that the change may negatively impact Nomad uses, so will address it in a future release. I think it's good to be transparent and proactively let users know that something may not work as intended, but we don't want to create the impression that we're not careful. Interested to hear other's opinions.

**Review by trujillo-adam — COMMENTED** [review]
I added a few comments, mostly for @boruszak

File: website/content/docs/upgrading/upgrade-specific.mdx
Comment (schmichael) [reply website/content/docs/upgrading/upgrade-specific.mdx:34]: But it is a bug.

File: website/content/docs/upgrading/upgrade-specific.mdx
Comment (schmichael) [reply website/content/docs/upgrading/upgrade-specific.mdx:31]: I'm open to anything. Just let me know if you want a callout or new heading or both.

**Comment by schmichael** [comment]
If somebody wants to fully takeover this PR and get it cleaned up and merged that's fine by me. It's not clear from the comments exactly what changes need to be made.

File: website/content/docs/upgrading/upgrade-specific.mdx
Comment (im2nguyen) [inline website/content/docs/upgrading/upgrade-specific.mdx:35]:
```suggestion
Consul 1.12.0 Enterprise introduced a change that prevents Nomad Enterprise from removing services from non-default Consul namespaces. To avoid errors, we recommend that Nomad Enterprise users wait to update Consul Enterprise until we fix this issue in a future release.
```

**Review by im2nguyen — APPROVED** [review]
lgtm!

**Comment by schmichael** [comment]
Thanks all! I don't have commit rights on this repo, so someone else will have to hit merge for me.

**Comment by blake** [comment]
Is this still waiting on a second approval from @trujillo-adam and @boruszak, or can it be merged?

File: website/content/docs/upgrading/upgrade-specific.mdx
Comment (trujillo-adam) [inline website/content/docs/upgrading/upgrade-specific.mdx:31]:
```suggestion
### Nomad Namespace Incompatibility

Nomad Enterprise users should not upgrade to Consul Enterprise 1.12.0. 

Consul 1.12.0 Enterprise introduced a change that prevents Nomad Enterprise from removing services from non-default Consul namespaces. To avoid errors, we recommend that Nomad Enterprise users wait to update Consul Enterprise until we fix this issue in a future release.
```
It may appear to readers that something is missing if we were to stack two H2s. I turned the second heading into a standalone paragraph so that it stands out with the explanation in the second paragraph.

**Review by trujillo-adam — CHANGES_REQUESTED** [review]

**Comment by schmichael** [comment]
Edits applied. Merge conflict resolved. Can someone please merge this? Nomad is receiving user reports about this bug.

**Review by kisunji — APPROVED** [review]

**Pull request merged**
